synapse/tests/common.py

```diff
@@ -25,6 +28,14 @@
 
 class TooFewEvents(Exception): pass
 
+# Py2/3 SSL Exception Compat
+if s_compat.version >= (3, 0, 0):
```

Comment:
funny - remcycle effectively is a "SynHTTPClient" but it eats exceptions like this and stores them in the job's results like any other exception.